Source/CsDebugScript.Engine/Process.cs

```diff
@@ -543,6 +544,25 @@ public int FindMemoryRegion(ulong address)
         }
 
         /// <summary>
+        /// Invalidates cache structures.
+        /// Use when memory state changes (e.g. during live debugging).
+        /// </summary>
+        public void InvalidateProcessCache()
```

Comment:
This function is really nice :)
Can you please change SimpleCache and DictionaryCache to implement new interface called ICache that provides a function to clear it?
That way, you can iterate here over both DictionaryCache and SimpleCache... And you can remove dynamic as you will now have interface to cast to.